etl/nifi_scripts/compliance_report.groovy

```diff
@@ -0,0 +1,647 @@
+// =========================================
+def sourceDbcpService = context.controllerServiceLookup.getControllerService('3245b078-0192-1000-ffff-ffffba20c1eb')
+def destinationDbcpService = context.controllerServiceLookup.getControllerService('3244bf63-0192-1000-ffff-ffffc8ec6d93')
+
+// =========================================
+// Main Execution
+// =========================================
+
+Connection sourceConn = null
+Connection destinationConn = null
+
+try {
+    sourceConn = sourceDbcpService.getConnection()
+    destinationConn = destinationDbcpService.getConnection()
+    destinationConn.setAutoCommit(false)
+
+    // Disable refresh of the materialized view
+    destinationConn.createStatement().execute('DROP FUNCTION IF EXISTS refresh_transaction_aggregate() CASCADE;')
+    destinationConn.createStatement().execute("""
+        CREATE OR REPLACE FUNCTION refresh_transaction_aggregate()
+        RETURNS void AS \$\$
+        BEGIN
+            -- Temporarily disable the materialized view refresh
+        END;
+        \$\$ LANGUAGE plpgsql;
+    """)
+
+    // Disable the refresh function
+  destinationConn.createStatement().execute('DROP FUNCTION IF EXISTS refresh_mv_director_review_transaction_count() CASCADE;')
+  destinationConn.createStatement().execute("""
+      CREATE OR REPLACE FUNCTION refresh_mv_director_review_transaction_count()
+      RETURNS void AS \$\$
+      BEGIN
+          -- Temporarily disable the materialized view refresh
+      END;
+      \$\$ LANGUAGE plpgsql;
+  """)
+
+    def referenceData = loadReferenceData(destinationConn)
+
+    // Fetch compliance reports and chains
+    def chains = fetchComplianceReports(sourceConn)
+
+    // Fetch compliance report history
+    def historyMap = fetchComplianceReportHistory(sourceConn)
+
+    // Fetch associated credit trades
+    def creditTradesMap = fetchApprovedCreditTrades(sourceConn)
+
+    // Prepare statements
+    def statements = prepareStatements(destinationConn)
+
+    int totalInserted = 0
+    int totalHistoryInserted = 0
+    int totalTransactionsInserted = 0
+
+    chains.each { rootId, chainReports ->
+        chainReports.sort { a, b ->
+            a.traversal <=> b.traversal ?: a.compliance_report_id <=> b.compliance_report_id
+        }
+
+        // Assign a single UUID for the chain
+        def groupUuid = UUID.randomUUID().toString()
+
+        def version = 0
+        chainReports.each { report ->
+            def currentStatusId = mapCurrentStatus(
+                report.fuel_supplier_status_id,
+                report.analyst_status_id,
+                report.manager_status_id,
+                report.director_status_id,
+                referenceData
+            )
+
+            def supplementalInitiator = report.is_supplemental ? "SUPPLIER_SUPPLEMENTAL" : null
+            def reportingFrequency = "ANNUAL"
+
+            // create_user and update_user are always strings
+            def reportCreateUser = "imported_user"
```

Comment:
updating the data as imported_user might cause issues w.r.t displaying user names in UI history log.

Reply:
yes good catch, i will update.